Automattic/wp-calypso
# Gutenberg: Replace Gutenlypso with Calypsoified iframe

Related P2 posts: p7jreA-26b-p2, p7jreA-24O-p2, p7jreA-25P-p2. 

#### Changes proposed in this Pull Request

* Replace Gutenlypso with Calypsoified iframe for Simple sites. The existing production behavior for Atomic, Jetpack, and VIP sites should remain unchanged.

* In order to simplify the code and flows, I decided to replace the `isGutenbergEnabled` with `isCalypsoifyGutenbergEnabled`. `isGutenbergEnabled` was previously used as a generic way to check both for Gutenlypso (Simple sites) and Calypsoify (Atomic). Since we are removing Gutenlypso now, we can boil this down to just Calypsoify checks. This also allows us to remove the `gutenberg` feature flag completely, which was controlling the Gutenlypso functionality so far.

* After this change, a lot of the Gutenlypso code will become unused. In order to keep this PR from exploding in size and make it easier to review, I intentionally didn't delete it. I'll make sure to clean up the remaining unused code in follow-up PRs.

#### Testing instructions

For Simple sites:

0. The behavior for Atomic, Jetpack, and VIP sites should match production.
1. Apply ~~D23405-code~~ and ~~D23473-code~~ to your sandbox.
2. Use sandboxed Simple site for your testing.
3. Verify that opt-in and opt-out flows still work - try clicking both the buttons available in Inline Help, and `Switch to Classic` in wp-admin.
4. Verify that Inline Help is working as expected.
5. Test various editor flows and make sure that everything still works (publishing, updating). Try with various post types.
6. Make sure that back button in Gutenberg editor is working properly - it should navigate the parent frame, and not in the iframe (note: redirects to `wordpress.com`, might be confusing while testing).
